lib/node_modules/@stdlib/array/float16/lib/polyfill.js

```diff
@@ -0,0 +1,903 @@
+    }
+});
+
+/**
+* Converts a float32 number to a float16 (binary) representation.
+*
+* @param {number} val - Float32 value
+* @returns {integer} Float16 bit pattern
+*/
+Float16Array.toFloat16 = function( val ) {
+    var floatView = new Float32Array( 1 );
+    var intView = new Uint32Array( floatView.buffer );
+    var sign;
+    var exponent;
+    var fraction;
+    var exp;
+    var frac;
+
+    floatView[ 0 ] = val;
+    var x = intView[ 0 ];
+
+    sign = ( x >> 16 ) & 0x8000;
+    exponent = ( ( x >> 23 ) & 0xff ) - 127;
+    fraction = x & 0x7fffff;
+
+    if ( exponent <= -15 ) {
+        if ( exponent < -24 ) {
+            return sign;
+        }
+        return sign | ( ( ( fraction | 0x800000 ) >> ( -exponent - 1 ) ) >> 13 );
+    }
+
+    if ( exponent >= 16 ) {
+        return sign | 0x7c00;
+    }
+
+    exp = exponent + 15;
+    frac = fraction >> 13;
+    return sign | ( exp << 10 ) | frac;
+};
+
+/**
+* Converts a float16 binary representation to float32.
+*
+* @param {integer} bits - Float16 bit pattern
+* @returns {number} Float32 value
+*/
+Float16Array.fromFloat16 = function( bits ) {
+    var sign = ( bits & 0x8000 ) << 16;
+    var exponent = ( bits & 0x7C00 ) >> 10;
+    var fraction = bits & 0x03FF;
+    var intView = new Uint32Array( 1 );
+    var floatView;
+    var result;
+    var exp32;
+    var frac32;
+
+    if ( exponent === 0 ) {
+        if ( fraction === 0 ) {
+            result = sign;
+        } else {
+            exponent = 1;
+            while ( ( fraction & 0x0400 ) === 0 ) {
+                fraction <<= 1;
+                exponent -= 1;
+            }
+            fraction &= ~0x0400;
+            exp32 = 127 - 15 - exponent;
+            frac32 = fraction << 13;
+            result = sign | ( exp32 << 23 ) | frac32;
+        }
+    } else if ( exponent === 0x1F ) {
+        result = sign | 0x7F800000 | ( fraction << 13 );
+    } else {
+        exp32 = exponent - 15 + 127;
+        result = sign | ( exp32 << 23 ) | ( fraction << 13 );
+    }
+
+    intView[ 0 ] = result;
+    floatView = new Float32Array( intView.buffer );
+    return floatView[ 0 ];
+};
+
+/**
+* Create a new Float16Array from a variable number of arguments.
+*
+* @returns {Float16Array} New instance
+*/
+Float16Array.of = function() {
+    var len = arguments.length;
+    var result = new Float16Array( len );
+    var i;
+
+    for ( i = 0; i < len; i++ ) {
+        result.set( i, arguments[ i ] );
+    }
+    return result;
+};
+
+/**
+* Create a new Float16Array from an array-like or iterable object.
+*
+* @param {*} source - Source to convert
+* @param {Function} [mapFn] - Optional mapping function
+* @param {*} [thisArg] - Execution context for mapFn
+* @returns {Float16Array} New instance
+*/
+Float16Array.from = function( source, mapFn, thisArg ) {
+    if ( source == null ) {
+        throw new TypeError( 'invalid argument. Source cannot be null or undefined. Value: `' + source + '`.' );
```

Comment:
For error formatting, as in `array/bool`, you should be using `@stdlib/string/format`.